compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -1325,6 +1325,17 @@ pub enum CastKind {
     IntToFloat,
     PtrToPtr,
     FnPtrToPtr,
+    /// Convert a raw pointer to an `impl Pointee<Metadata = M>` into `M`.
+    ///
+    /// For example, this will give a `()` from `*const i32`, a `usize` from
+    /// `*mut [u8]`, or a vtable from a `*const dyn Foo`.
+    ///
+    /// There's only one legal cast type based on the input type, but calculating
+    /// that type is expensive, and thus it's convenient for this to be a `Cast`
+    /// instead of an `UnOp`.
```

Comment:
```suggestion
    /// that type is expensive, and thus it's convenient for this to be a `Cast`
    /// (that stores the output type in MIR) instead of an `UnOp`.
```

Reply:
I'm still a bit skeptical a cast for this. A dedicated `Rvalue` variant would be easier to understand, and consistent with the current `Rvalue::Len`. And if it needs to cache the destination type, so be it.